docs/source/wrappers.rst

```diff
@@ -0,0 +1,487 @@
+
+    clf = KerasClassifier(model=model_build_fn, optimizer=Adam)
+    clf.fit(X, y)
+    y_pred = clf.predict(X_valid)
+
+
+Arguments to ``model``/``model_build_fn``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You probably wish to pass parameters from :py:class:`~scikeras.wrappers.BaseWrapper`
+to ``model``, or you may want to use attributes from
+:py:class:`~scikeras.wrappers.BaseWrapper` such as ``n_features_in_`` while building
+your model. SciKeras allows you to do both.
+
+To enable this, SciKeras uses two special arguments to ``model`` that will only
+be passed if they are present in ``model``'s signature (i.e. there is an argument
+with the same name in ``model``'s signature):
+
+``meta_params``
++++++++++++++++
+This is a dictionary containing all of the attributes that
+:py:class:`~scikeras.wrappers.BaseWrapper` creates when it is initialized
+These include ``n_features_in_``, ``y_dtype_``, etc. For a full list,
+see the :ref:`scikeras-api` documentation.
+
+``compile_kwargs``
+++++++++++++++++++
+This is a dictionary of parameters destined for :py:func:`tensorflow.Keras.Model.compile`.
+You will want to accept this parameter unless you are returning an un-compiled ``Model``
+instance. Parameters available via this dictionary are:
+
+* ``optimizer``
+* ``loss``
+* ``callbacks``
+* Any other parameters with the prefix ``optimizer__``, ``loss__`` or ``callbacks__``
+
+Keyword arguments to :py:func:`scikeras.wrappers.BaseWrapper.__init__`
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+Any **other** keyword arguments passed to :py:func:`scikeras.wrappers.BaseWrapper.__init__` when
+the wrapper is instantiated will also be passed to ``model_build_fn`` as keyword
+arguments. For example, calling ``KerasClassifier(myparam=10)`` will result in a
+``model_build_fn(my_param=10)`` call.
+
+
+Arguments to :py:class:`scikeras.wrappers.BaseWrapper`
+------------------------------------------------------
+
+A complete explanation of all arguments and methods of
+:py:class:`.BaseWrapper` are found in the :ref:`scikeras-api` documentation. Here we
+focus on the main ones.
+
+model
+^^^^^
+
+This is where you pass your Keras ``Model``.
+Ideally, it should not be instantiated, although instantiated models
+are accepted. Instead, the init arguments
+for your model should be passed to :py:class:`.BaseWrapper` either as
+bare keyword arguments (if their name does not conflict with any of
+the existing keyword arguments) or with the ``model__`` prefix,
+which will override any arguments of the same name only when passed to
+``model``. E.g., if your module takes the arguments
+``hidden_layer_sizes`` and ``lr``, the code would look like this:
+
+.. code:: python
+
+    def model_build_fn(hidden_layer_sizes, lr):
+        model = Model(...)
+        ...
+        return model
+
+    clf = KerasClassifier(
+        model=model_build_fn,
+        hidden_layer_sizes=(100,),
```

Comment:
I kind of agree, but I also think that there is nowhere else those parameters could go, so I don't think there is much harm in keeping around this "syntactic sugar" of being able to pass parameters without `model__`.

To take it one step further, what if we did the opposite, and didn't even use the `model__` prefix? Since there are no "default" parameters being passed to `model`, there is nothing to override, so I don't see the point of the prefix?

I know that Skorch uses the prefix, but we don't need to be _exactly_ like them.

Reply:
> The only con I would see is that it would make it impossible to add parameters via subclassing, which is pretty important.

I think this is pretty important and a dealbreaker, and so I am +1 on forcing the use of `model__`.

Reply:
I'm also for forcing the user to use `model__`, but for separate reasons. I think the consistent interface will be easier to document and implement.

I think there should be a deprecation period with some warnings.

Reply:
> I think there should be a deprecation period with some warnings.

Luckily I don't think this is necessary. SciKeras introduced the ability to pass parameters to `build_fn`, so this is only breaking the SciKeras API, not the old API from tensorflow (which I am more cautious to break).